src/ngraph/op/experimental/reshape.hpp

```diff
@@ -0,0 +1,65 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/node.hpp"
+#include "ngraph/op/op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v1
+        {
+            /// \brief Tensor dynamic reshape operation.
+            ///
+            /// "Converts" an input tensor into a new shape with the same number of elements.
+            /// This op does not touch the actual data. If needed, use Transpose for that purpose.
+            ///
+            class Reshape : public Op
+            {
+            public:
+            public:
+                NGRAPH_API
+                static constexpr NodeTypeInfo type_info{"DynReshape", 1};
```

Comment:
`DynReshape` => `Reshape`

Reply:
It was discussed earlier (for `ReduceSum` and `ReduceProd`). In current implementation `get_typeid` mechanism is based on op description (it is a key in map where the value is `OP_TYPEID` - enum represents op). When we change the description to `Reshape` using `switch` will be not working correctly. The conclusion of preview discussion was that it should be fixed in separate PR.

Reply:
I agree here---there's a higher-order issue with this stuff we'll need to figure out.